spec/services/checksum_validator_spec.rb

```diff
@@ -1,7 +1,126 @@
-require 'rails_helper'
+require_relative '../../app/services/checksum_validator.rb'
+require_relative '../load_fixtures_helper.rb'
 
 RSpec.describe ChecksumValidator do
-  it 'has some behaviors' do
-    skip 'we should figure out what they are and test them'
+  let(:storage_dir) { "spec/fixtures/checksum_root01/moab_storage_trunk" }
+  let(:endpoint) { Endpoint.find_by(storage_location: storage_dir) }
+  let(:object_dir) { "#{storage_dir}/#{DruidTools::Druid.new(druid).tree.join('/')}" }
+
+  context '#initialize' do
+    it 'sets attributes' do
+      druid = 'bj102hs9687'
+      cv = described_class.new(druid, storage_dir)
+      expect(cv.druid).to eq "druid:#{druid}"
+      expect(cv.endpoint).to eq endpoint
+      expect(cv.handler_results).to be_an_instance_of AuditResults
+    end
+  end
+
+  context '#validate_manifest_inventories' do
+    let(:druid) { 'bj102hs9687' }
+    let(:cv) { described_class.new(druid, storage_dir) }
+
+    it 'instantiates storage_object from druid and druid_path' do
+      expect(Moab::StorageObject).to receive(:new).with(cv.druid, a_string_matching(object_dir)).and_call_original
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls validate_manifest_inventory for each storage_object_version' do
+      sov1 = instance_double(Moab::StorageObjectVersion)
+      sov2 = instance_double(Moab::StorageObjectVersion)
+      sov3 = instance_double(Moab::StorageObjectVersion)
+      version_list = [sov1, sov2, sov3]
+      storage_object = instance_double(Moab::StorageObject, version_list: [sov1, sov2, sov3])
+      allow(cv).to receive(:storage_object).and_return(storage_object)
+      version_list.each do |storage_object_version|
+        expect(cv).to receive(:validate_manifest_inventory).with(storage_object_version)
+      end
+      cv.validate_manifest_inventories
+    end
+
+    it 'calls AuditResults.report_results' do
+      results = instance_double(AuditResults, add_result: nil, :actual_version= => nil, :check_name= => nil)
+      allow(AuditResults).to receive(:new).and_return(results)
+      expect(results).to receive(:report_results)
+      cv.validate_manifest_inventories
+    end
+
+    context 'modify file signature in manifestInventory.xml' do
```

Comment:
er ... is this really the context of   "file signature in manifestInventory.xml doesn't validate" ?    Because in the larger context of this check, we aren't modifying anything -- we're validating checksums.

Also, we want to be certain this happens for both a metadata file and for a content file.